app/models/name/taxonomy.rb

```diff
@@ -523,26 +532,38 @@ def propagate_classification
   # names below genus with the same generic epithet.  Then add all those
   # names' synonyms.
   def subtaxa_whose_classification_needs_to_be_changed
-    subtaxa = Name.where("deprecated IS FALSE AND " \
-                         "text_name LIKE ?",
-                         "#{text_name} %").to_a
-    synonyms = Name.where("deprecated IS TRUE AND " \
-                          "synonym_id IN (?) AND " \
-                          "classification != ?",
-                          subtaxa.map(&:synonym_id).reject(&:nil?).uniq,
-                          classification)
+    # subtaxa = Name.where("deprecated IS FALSE AND " \
+    #                      "text_name LIKE ?",
+    #                      "#{text_name} %").to_a
+    subtaxa = Name.where(Name[:deprecated].eq(Arel.sql("FALSE")).
+                         and(Name[:text_name].matches("#{text_name} %"))).to_a
+    # synonyms = Name.where("deprecated IS TRUE AND " \
+    #                       "synonym_id IN (?) AND " \
+    #                       "classification != ?",
+    #                       subtaxa.map(&:synonym_id).reject(&:nil?).uniq,
+    #                       classification)
+    synonyms = Name.where(Name[:deprecated].eq(Arel.sql("TRUE")).
+                          and(Name[:synonym_id].in(
+                                subtaxa.map(&:synonym_id).reject(&:nil?).uniq
+                              )).
+                          and(Name[:classification].not_eq(classification)))
     (subtaxa + synonyms).map(&:id).uniq
   end
 
   # This is meant to be run nightly to ensure that all the classification
   # caches are up to date.  It only pays attention to genera or higher.
   def self.refresh_classification_caches
```

Comment:
Oh geez, I am about to expire.  That's all I can handle in one sitting, sorry. :(